src/utils/hooks/query/useStudentScoreQuery.ts

```diff
@@ -5,7 +5,7 @@ import { User } from '../../../types';
 
 export const QUERY_KEY_USER_SCORE = 'user-scores';
 
-export const UseStudentScoreQuery = (user: User) =>
+export const UseStudentScoreQuery = (user?: User | null) =>
```

Comment:
мне не нравится, что типизация настроена как-то так, что приходится объявлять user необязательным
Хотя, в этом методе этот аргумент надо передавать обязательно
Это он может быть пустым просто
Может, лучше так?
```suggestion
export const UseStudentScoreQuery = (user: User | undefined) =>
```

так аргумент будет обязательный и не ts будет ругаться на такое
```ts
UseStudentScoreQuery()
```
но при этом внутри хука он будет предупреждать, что он может быть пустым